osutil/disks/luks.go

```diff
@@ -53,8 +53,8 @@ func cryptLuks2DeviceMapperBackResolver(dmUUID, dmName []byte) (dev string, ok b
 	// we are extra safe here since the dm name could be hypothetically user
 	// controlled via an external USB disk with LVM partition names, etc.
 	dmUUIDSafe := bytes.TrimSuffix(
-		bytes.TrimSpace(dmUUID),
-		append([]byte("-"), bytes.TrimSpace(dmName)...),
+		bytes.TrimSpace(opts.dmUUID),
```

Comment:
most of the function could be extracted into a helper shared by cryptLuks2DeviceMapperBackResolver and cryptVerityDeviceMapperBackResolver if instead of using ID_FS_UUID we simply use the UUID from device mapper

Reply:
@bboozzoo I left a more extensive comment that also relates to this in the previous comment but in short I think the verity resolver won't be able to use the same field (`DM_UUID`) to do what it needs to do so maybe this comment doesn't apply anymore.